File: P4/Civitas/src/civitas/SorpresaCasilla.java
package civitas;
import java.util.ArrayList;
/**
 *
 * @author abel
 */
public class SorpresaCasilla extends Sorpresa {
    
    private Tablero tablero;
    private int valor;
    
    SorpresaCasilla(Tablero tablero, int valor, String texto){
        this.tablero = tablero;
        this.valor = valor;
        this.texto = texto;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){ // Revisar
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            int casilla = todos.get(actual).getNumCasillaActual();
            int tirada = tablero.calcularTirada(casilla, valor);
            int n = tablero.nuevaPosicion(casilla, tirada);
            todos.get(actual).moverACasilla(n);
            tablero.getCasilla(n).recibeJugador(actual, todos);
        }
    }
    
}

File: P4/civitasruby/lib/civitas/sorpresa_casilla.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

# Salva
module Civitas
  class Sorpresa_casilla < Sorpresa  # sorpresa que manda a una casilla al jugador
    def initialize(tablero, valor, texto)
      super(texto)
      @tablero = tablero
      @valor = valor
    end
    
    def aplicar_a_jugador(actual,todos)
      casilla = -1
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        casilla = todos.at(actual).get_num_casilla_actual
        tirada = @tablero.calcular_tirada(casilla, @valor)
        nueva = @tablero.nueva_posicion(casilla,tirada)
        todos.at(actual).mover_a_casilla(nueva)
        @tablero.get_casilla(nueva).recibe_jugador(actual,todos)
      end
    end
    
    
  end
end
